File: week_6/sentimental-readability/readability.py
def main():
    text = str(input('Text: '))  # Getting input
    print(get_grade(text))  # Printing grade


def get_grade(t):
    s = 0
    w = 1
    l = 0

    # Couting Sentences, Words and Letters
    for i in range(len(t)):
        if t[i] in ('.', '!', '?'):
            s += 1
        if t[i] == ' ':
            w += 1
        if t[i].isalpha():
            l += 1

    # Getting the average
    S = float(s * 100 / w)
    L = float(l * 100 / w)

    # Calculating grade
    grade = round(0.0588 * L - 0.296 * S - 15.8)

    # Checking grade
    if grade < 1:
        return 'Before Grade 1'
    elif grade >= 16:
        return 'Grade 16+'
    else:
        return f'Grade {grade}'


if __name__ == '__main__':
    main()

File: week_2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// Setting function
void get_grade(string s);

int main(void)
{
    // Inputting the text
    string text = get_string("Text: ");
    // Computing the grade
    get_grade(text);
}

void get_grade(string s)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    // Counting the nubmer of letters, words and sentences
    for (int i = 0; i < strlen(s); i++)
    {
        if (isalpha(s[i]))
        {
            letters++;
        }
        if (isspace(s[i]))
        {
            words++;
        }
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentences++;
        }
    }
    // Calculating the grade
    float L = (float)letters * 100 / words;
    float S = (float)sentences * 100 / words;
    int grade = round(0.0588 * L - 0.296 * S - 15.8);
    // Printing the grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}